sygnal/apnspushkin.py

```diff
@@ -242,7 +242,10 @@ async def _dispatch_request(
         log.info(f"Sending as APNs-ID {notif_id}")
         span.set_tag("apns_id", notif_id)
 
-        device_token = base64.b64decode(device.pushkey).hex()
+        if self.get_config("apns_device_token_use_hex", bool, True):
```

Comment:
```suggestion
        # Some client libraries will provide the push token in hex format already. Avoid
        # attempting to convert from base 64 to hex.
        if self.get_config("apns_device_token_use_hex", bool, True):
```

Or something like that.